src/NuGet.Indexing/DownloadsByVersion.cs

```diff
@@ -48,8 +41,13 @@ public int Total
 
             set
             {
+                if (_downloadsByVersion.ContainsKey(version))
+                {
+                    _total = _total - _downloadsByVersion[version];
```

Comment:
It sounds like you're saying we should separate the HTTP 500 bug fix from addressing the data consistency issue that Scott is referring to. This seems reasonable to me. I don't see  a problem in fixing one thing at a time, as long as the follow-up work is tracked and properly prioritized/scheduled.